src/configure/templateInputHelper/InputControlProvider.ts

```diff
@@ -63,6 +73,50 @@ export class InputControlProvider {
         this._setInputControlDataSourceInputs();
     }
 
+    private async setInputControlValueFromRepoAnalysisResult(inputControl: InputControl): Promise<void> {
+        let repoAnalysisSettingKey = inputControl.getPropertyValue(this.repoAnalysisSettingKey);
+        if (this._repoAnalysisSettingInUse !== -1) {
+            if (this._repoAnalysisSettings.length === 0 || !this._repoAnalysisSettings[this._repoAnalysisSettingInUse].settings[repoAnalysisSettingKey]) {
+                let value = await new ControlProvider().showInputBox(repoAnalysisSettingKey, { placeHolder: inputControl.getInputDescriptor().name });
+                inputControl.setValue(value);
+            } else {
+                inputControl.setValue(this._repoAnalysisSettings[this._repoAnalysisSettingInUse].settings[repoAnalysisSettingKey]);
+            }
+        } else {
+            let settingIndexMap: Map<string, number[]> = new Map();
+            this._repoAnalysisSettings.forEach((analysisSetting, index: number) => {
+                if (!analysisSetting.settings[repoAnalysisSettingKey]) {
+                    return;
+                }
+                if (settingIndexMap.has(analysisSetting.settings[repoAnalysisSettingKey])) {
+                    settingIndexMap.get(analysisSetting.settings[repoAnalysisSettingKey]).push(index);
+                } else {
+                    settingIndexMap.set(analysisSetting.settings[repoAnalysisSettingKey], [index]);
+                }
+            });
+            if (settingIndexMap.size === 0) {
+                let value = await new ControlProvider().showInputBox(repoAnalysisSettingKey, { placeHolder: inputControl.getInputDescriptor().name });
```

Comment:
please log a trace here as ideally this is not expected

Reply:
done